### PR TITLE
fix setting code units in PlutoDataset

### PR DESCRIPTION
**Pull request by xshaokun**
Reminded by the discussion with @dutta-alankar in #202, some bugs need to be fixed.

- [x] adopt the correct default code unit values in Pluto
- [x] add the parsers for `log` and `log10` in unit definitions

### CODE REVIEW

**Comment by dutta-alankar** [comment]
Please have a look [here](https://github.com/neutrinoceros/yt_idefix/pull/202#discussion_r1117339123)

**Comment by xshaokun** [comment]
Hi @dutta-alankar, as suggest by @neutrinoceros , let's continue our discussion about your `PlutoXdmfDataset._set_code_unit_attributes(self)` here.

I've reviewed your method carefully, and I believe that the existing method `PlutoVtkDataset._set_code_unit_attributes(self)` are able to do all things that your method want to do after this PR (refer to `PlutoVtkDataset._set_code_unit_attributes(self)` and `PlutoVtkDataset.__parse_definitions_header(self)`), so you don't need to override it. If I missed anything, please tell me so that we can know what you can do in the existing methods.

FYI, here I list the PRs relating to parsing unit definitions for Pluto, i.e. #56 , #69 and #167. If you have any question about the mechanism how `PlutoVtkDataset` parses the unit definitions, feel free to ask me here :)

**Comment by dutta-alankar** [comment]
> Hi @dutta-alankar, as suggest by @neutrinoceros , let's continue our discussion about your `PlutoXdmfDataset._set_code_unit_attributes(self)` here.
> 
> I've reviewed your method carefully, and I believe that the existing method `PlutoVtkDataset._set_code_unit_attributes(self)` are able to do all things that your method want to do after this PR (refer to `PlutoVtkDataset._set_code_unit_attributes(self)` and `PlutoVtkDataset.__parse_definitions_header(self)`), so you don't need to override it. If I missed anything, please tell me so that we can know what you can do in the existing methods.
> 
> FYI, here I list the PRs relating to parsing unit definitions for Pluto, i.e. #56 , #69 and #167. If you have any question about the mechanism how `PlutoVtkDataset` parses the unit definitions, feel free to ask me here :)

@xshaokun Thanks a lot for your inputs. I'll try to use your implementation of the code by inheriting methods from the parent class of PlutoXdmfDataset. I'll drop comments if I'm stuck with something regarding that. 

On a separate note, you might need to put these codes in class ```PlutoStaticDataset``` later on as this would be the parent class for ```PlutoVtkDataset``` and ```PlutoXdmfDataset```. Please see the discussion here: [link](https://github.com/neutrinoceros/yt_idefix/pull/202#issuecomment-1445006296).

**Comment by xshaokun** [comment]
> On a separate note, you might need to put these codes in class `PlutoStaticDataset` later on as this would be the parent class for `PlutoVtkDataset` and `PlutoXdmfDataset`. Please see the discussion here: [link](https://github.com/neutrinoceros/yt_idefix/pull/202#issuecomment-1445006296).

That's great! I'm willing to do it, or you can just do it in your PR if you want.

**Comment by dutta-alankar** [comment]
Also I'm very new to using so many features on GitHub. Can you also generate a PR on my fork so that I can use your updated code right now and use the inheritance you provided before your PR gets accepted by @neutrinoceros . However, I can also wait if you suggest doing so until your PR gets merged.

**Comment by xshaokun** [comment]
@dutta-alankar, I estimate this PR will be merged soon, so I suggest you can wait until it's merged. You don't have problems about rebasing your branch, do you?

**Comment by neutrinoceros** [comment]
>  I estimate this PR will be merged soon, so I suggest you can wait until it's merged

That's the plan.
@dutta-alankar please provide input here if you need to.

**Comment by dutta-alankar** [comment]
> @dutta-alankar, I estimate this PR will be merged soon, so I suggest you can wait until it's merged. You don't have problems about rebasing your branch, do you?

Ok I'll wait for the merge. I think rebasing won't be an issue.

**Review by neutrinoceros — APPROVED** [review]
Looks good, thanks again @xshaokun !

File: yt_idefix/data_structures.py
Comment (dutta-alankar) [inline yt_idefix/data_structures.py:601]: @xshaokun Can you also add a replacement of `pow` with `np.power` as well?
What is very common is that for example masses are put in say `pluto.ini` as 13.5 and in defeinitions then it is used as pow(10.,13.5)*CONST_Msun.

Comment (dutta-alankar) [inline yt_idefix/data_structures.py:601]: @xshaokun Can you also add a replacement of `pow` with `np.power` as well?
What is very common is that for example masses are put in say `pluto.ini` as 13.5 and in defeinitions then it is used as pow(10.,13.5)*CONST_Msun.
```suggestion
               expr = re.sub(r"log10", "np.log10", expr)
               expr = re.sub(r"pow", "np.power", expr)
```

Comment (xshaokun) [reply yt_idefix/data_structures.py:601]: Ah because there's no problem to execute `pow(10, 13.5)` in Python, I think there's no need to add more code to replace it.

**Comment by neutrinoceros** [comment]
@xshaokun I'm not yet sure how long it'll take to integrate the XDMF PR, so would you like me to cut a bugfix release with just your patch ?

File: yt_idefix/data_structures.py
Comment (xshaokun) [reply yt_idefix/data_structures.py:601]: > Also wanted to ask that will this not happen?
> 
> ```
> log10 -> np.log10
> ```
> 
> by line `600` and then
> 
> ```
> np.log10 -> np.np.log10
> ```
> 
> by line `601` creating problem Perhaps changing `log` to `np.log` will automatically account for `log10` as well.

Nice catch! I was really careless about this. Let's fix it right away. Thanks!

**Comment by xshaokun** [comment]
> @xshaokun I'm not yet sure how long it'll take to integrate the XDMF PR, so would you like me to cut a bugfix release with just your patch ?

@neutrinoceros I'm okay about that because I installed yt_idefix with editable mode, so the release is not in hurry. However, @dutta-alankar just caught a bug in this PR. Would you mind to revert  PR merge? Or I create anther PR to fix it.

File: yt_idefix/data_structures.py
Comment (xshaokun) [reply yt_idefix/data_structures.py:601]: @dutta-alankar Could you provide me a small patch of test data which includes log and log10 in unit definitions? I'll add them in test/data and you can also use them in the future.

Comment (xshaokun) [reply yt_idefix/data_structures.py:601]: > @dutta-alankar Could you provide me a small patch of test data which includes log and log10 in unit definitions? I'll add them in test/data and you can also use them in the future.

Ah never mind. Currently we are only able to test vtk outputs. I'll try to modify the existing test data.

**Comment by neutrinoceros** [comment]
If it's not too much trouble I'd prefer you open a second PR

**Pull request merged**
